content/en/blog/_posts/2020-12-16-third-party-device-metrics-hits-ga.md

```diff
@@ -0,0 +1,134 @@
+
+For example, the NVIDIA GPU dcgm-exporter:
+
+```
+$ curl -sL http://127.0.01:8080/metrics
+
+
+# HELP DCGM_FI_DEV_SM_CLOCK SM clock frequency (in MHz).
+# TYPE DCGM_FI_DEV_SM_CLOCK gauge
+# HELP DCGM_FI_DEV_MEM_CLOCK Memory clock frequency (in MHz).
+# TYPE DCGM_FI_DEV_MEM_CLOCK gauge
+# HELP DCGM_FI_DEV_MEMORY_TEMP Memory temperature (in C).
+# TYPE DCGM_FI_DEV_MEMORY_TEMP gauge
+...
+DCGM_FI_DEV_SM_CLOCK{gpu="0", UUID="GPU-604ac76c-d9cf-fef3-62e9-d92044ab6e52",container="foo",namespace="bar",pod="baz"} 139
+DCGM_FI_DEV_MEM_CLOCK{gpu="0", UUID="GPU-604ac76c-d9cf-fef3-62e9-d92044ab6e52",container="foo",namespace="bar",pod="baz"} 405
+DCGM_FI_DEV_MEMORY_TEMP{gpu="0", UUID="GPU-604ac76c-d9cf-fef3-62e9-d92044ab6e52",container="foo",namespace="bar",pod="baz"} 9223372036854775794
+```
+
+Each agent is expected to adhere to the node monitoring guidelines. In other words, plugins are expected to generate prometheus metrics and new metrics should not have any dependency on the Kubernetes base directly.
+
+This allows consumers of the metrics to use a compatible monitoring pipeline to collect and analyze metrics from a variety of agents, even if they are maintained by different vendors.
+
+![Device metrics flowchart](/images/blog/2020-12-10-third-party-device-metrics-hits-ga/metrics-chart.png)
+
+## Disabling the NVIDIA GPU metrics - Warning 
+
+With the graduation of the plugin monitoring system, Kubernetes is deprecating the NVIDIA GPU metrics that are being reported by the kubelet.
+
+With the [DisableAcceleratorMetrics](/docs/concepts/cluster-administration/system-metrics/#disable-accelerator-metrics) feature being enabled by default in Kubernetes 1.20, NVIDIA GPUs will no longer be special citizens  in Kubernetes. This is a good thing in the spirit of being vendor neutral and enabling the best suited people to maintain their plugin on their own release schedule!
+
+Users will now need to either install the [NVIDIA dcgm-exporter](https://github.com/NVIDIA/gpu-monitoring-tools) or [use bindings](https://github.com/nvidia/go-nvml) to gather more accurate and complete metrics about NVIDIA GPUs. They can no longer rely on metrics that were reported by kubelet, such as `nvidia_gpu_duty_cycle` or `nvidia_gpu_memory_used_bytes` which were used to gather NVIDIA GPU memory utilization.
+
+This means that users who used to rely on the NVIDIA GPU metrics reported by kubelet, will need to update their reference and deploy the NVIDIA plugin. Namely the different metrics reported by kubernetes map to the following metrics:
+
+| Kubernetes Metrics              | NVIDIA dcgm-exporter metric                 |
+| ------------------------------- | ------------------------------------------- |
+| `nvidia_gpu_duty_cycle`         | `DCGM_FI_DEV_GPU_UTIL`                      |
```

Comment:
Where does the nvidia_gpu_duty_cycle come from?  The cAdvisor one is named container_accelerator_duty_cycle: https://github.com/google/cadvisor/blob/master/metrics/prometheus.go#L520

Reply:
This is correct, metrics should be:
- `container_accelerator_memory_total_bytes`
- `container_accelerator_memory_used_bytes`
- `container_accelerator_duty_cycle`

Should have spinned an instance, thanks for pointing them out!

Reply:
Sounds good, I'll fix it right now